Src/Newtonsoft.Json/Linq/JPropertyKeyedCollection.cs

```diff
@@ -189,7 +191,7 @@ protected override void SetItem(int index, JToken item)
             }
         }
 
-        public bool TryGetValue(string key, out JToken value)
+        public bool TryGetValue(string key, [NotNullWhen(true)]out JToken? value)
```

Comment:
I've added the `using System.Diagnostics.CodeAnalysis` to my own code but it's not finding this `NotNullWhenAttribute` in VS 2019 16.3 Preview 1. Do you know what could be the issue?

Reply:
I don't know. I use internal copies in this solution.

Reply:
These were added to corefx a couple of months ago as part of preview 6: 
https://github.com/dotnet/corefx/pull/38020
https://github.com/dotnet/coreclr/pull/24679

https://github.com/dotnet/core/blob/master/release-notes/3.0/preview/api-diff/preview6/3.0-preview6_System.Diagnostics.CodeAnalysis.md

What version of the dotnet sdk do you have (if you run `donet --version` on the command line, you will see it)? I believe VS 16.3 preview 2+ has the newer SDK that probably has the change you are looking for. Alternatively, you can install the latest preview sdk independently of VS (at preview 7 now), or the nightly:
https://dotnet.microsoft.com/download/dotnet-core/3.0
https://github.com/dotnet/core-sdk/blob/master/README.md#installers-and-binaries